internal/beater/api/mux.go

```diff
@@ -116,18 +117,18 @@ func NewMux(
 		handlerFn func() (request.Handler, error)
 	}
 
-	otlpHandlers := otlp.NewHTTPHandlers(zapLogger, batchProcessor, semaphore)
-	rumIntakeHandler := builder.rumIntakeHandler()
+	otlpHandlers := otlp.NewHTTPHandlers(zapLogger, batchProcessor, semaphore, meterProvider)
+	rumIntakeHandler := builder.rumIntakeHandler(meterProvider)
 	routeMap := []route{
-		{RootPath, builder.rootHandler(publishReady)},
-		{AgentConfigPath, builder.backendAgentConfigHandler(fetcher)},
-		{AgentConfigRUMPath, builder.rumAgentConfigHandler(fetcher)},
+		{RootPath, builder.rootHandler(publishReady, meterProvider)},
+		{AgentConfigPath, builder.backendAgentConfigHandler(fetcher, meterProvider)},
+		{AgentConfigRUMPath, builder.rumAgentConfigHandler(fetcher, meterProvider)},
 		{IntakeRUMPath, rumIntakeHandler},
 		{IntakeRUMV3Path, rumIntakeHandler},
-		{IntakePath, builder.backendIntakeHandler},
-		{OTLPTracesIntakePath, builder.otlpHandler(otlpHandlers.HandleTraces, otlp.HTTPTracesMonitoringMap)},
-		{OTLPMetricsIntakePath, builder.otlpHandler(otlpHandlers.HandleMetrics, otlp.HTTPMetricsMonitoringMap)},
-		{OTLPLogsIntakePath, builder.otlpHandler(otlpHandlers.HandleLogs, otlp.HTTPLogsMonitoringMap)},
+		{IntakePath, builder.backendIntakeHandler(meterProvider)},
+		{OTLPTracesIntakePath, builder.otlpHandler(otlpHandlers.HandleTraces, "apm-server.otlp.http.traces.", meterProvider)},
+		{OTLPMetricsIntakePath, builder.otlpHandler(otlpHandlers.HandleMetrics, "apm-server.otlp.http.metrics.", meterProvider)},
+		{OTLPLogsIntakePath, builder.otlpHandler(otlpHandlers.HandleLogs, "apm-server.otlp.http.logs.", meterProvider)},
```

Comment:
Why are you passing the `metricsPrefix` as an argument to the otel routes vs. hardcode them inside the handler functions for non-otel?